connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/ProcessInstanceContext.java

```diff
@@ -18,9 +18,43 @@
 
 import java.util.Map;
 
+/**
+ * Represents the context for a specific process instance. This interface provides a contract to
+ * interact with, modify, and retrieve information related to a particular instance of a process.
+ */
 public interface ProcessInstanceContext {
 
+  /**
+   * Retrieves the unique key associated with the process instance.
+   *
+   * @return A unique identifier representing the process instance.
+   */
   Long getProcessInstanceKey();
 
-  Map<String, Object> getVariables();
+  /**
+   * Updates the context with new or modified variables. This method provides a way to make changes
+   * to the state of the process instance.
+   *
+   * @param variables A map containing variable names as keys and their corresponding values.
+   *     Existing variables with the same name will be overwritten.
+   */
+  void update(Map<String, String> variables);
+
+  /**
+   * Determines if the current context is ready to bind properties. This can be used to check
+   * prerequisites or conditions before attempting to bind properties or perform further actions.
+   *
+   * @return {@code true} if the context is ready for binding, otherwise {@code false}.
+   */
+  boolean isReadyToBind();
+
+  /**
+   * Binds properties into a specified class type. This method allows for deserialization and
+   * conversion of properties into a given data structure or object.
+   *
+   * @param <T> The type into which properties should be deserialized.
+   * @param cls The class type that provides a blueprint for the deserialization.
+   * @return An instance of type {@code T} containing deserialized and bound properties.
+   */
+  <T> T bindProperties(Class<T> cls);
```

Comment:
Should be called just bind as it combines the process variables and the properties.